include/exadg/operators/variable_coefficients.h

```diff
@@ -23,168 +23,213 @@
+  }
 
 private:
+  /**
+   * Reinitializes the tables in use. This resizes the tables and initializes the coefficients with
+   * the default constructed @p coefficient_type.
+   */
   template<int dim, typename Number>
   void
   reinit(dealii::MatrixFree<dim, Number> const & matrix_free, unsigned int const quad_index)
   {
     coefficients_cell.reinit(matrix_free.n_cell_batches(), matrix_free.get_n_q_points(quad_index));
 
-    coefficients_face.reinit(matrix_free.n_inner_face_batches() +
-                               matrix_free.n_boundary_face_batches(),
-                             matrix_free.get_n_q_points_face(quad_index));
-
-    coefficients_face_neighbor.reinit(matrix_free.n_inner_face_batches(),
-                                      matrix_free.get_n_q_points_face(quad_index));
-
-    // TODO
-    // // cell-based face loops
-    // coefficients_face_cell_based.reinit(matrix_free.n_cell_batches()*2*dim,
-    // matrix_free.get_n_q_points_face(quad_index));
+    if(use_faces)
+    {
+      coefficients_face.reinit(matrix_free.n_inner_face_batches() +
+                                 matrix_free.n_boundary_face_batches(),
+                               matrix_free.get_n_q_points_face(quad_index));
+      coefficients_face_neighbor.reinit(matrix_free.n_inner_face_batches(),
+                                        matrix_free.get_n_q_points_face(quad_index));
+
+      if(use_cell_based_faces)
+      {
+        unsigned int const n_faces_per_cell =
+          matrix_free.get_dof_handler().get_triangulation().get_reference_cells()[0].n_faces();
+
+        coefficients_face_cell_based.reinit(matrix_free.n_cell_batches() * n_faces_per_cell,
+                                            matrix_free.get_n_q_points_face(quad_index));
+      }
+    }
   }
 
+  /**
+   * Fills the tables with a constant coefficient.
+   */
   void
   fill(coefficient_type const & constant_coefficient)
   {
     coefficients_cell.fill(constant_coefficient);
-    coefficients_face.fill(constant_coefficient);
-    coefficients_face_neighbor.fill(constant_coefficient);
 
-    // TODO
-    // coefficients_face_cell_based.fill(constant_coefficient);
-  }
+    if(use_faces)
+    {
+      coefficients_face.fill(constant_coefficient);
+      coefficients_face_neighbor.fill(constant_coefficient);
 
-  // variable coefficients
+      if(use_cell_based_faces)
+        coefficients_face_cell_based.fill(constant_coefficient);
+    }
+  }
 
-  // cell
+  //! Coefficient table for cells
   dealii::Table<2, coefficient_type> coefficients_cell;
 
-  // face-based loops
+  //! Coefficient table for faces
   dealii::Table<2, coefficient_type> coefficients_face;
+
+  //! Coefficient table for neighbor faces
   dealii::Table<2, coefficient_type> coefficients_face_neighbor;
 
-  // TODO
-  //  // cell-based face loops
-  //  dealii::Table<2, coefficient_type> coefficients_face_cell_based;
+  //! Coefficient table for faces with cell-based access
```

Comment:
Maybe add a comment that this is an "additional" data structure (in addition to the face coefficients above), because `use_cell_based_face_loops` is additional/optional functionality. See also comment https://github.com/exadg/exadg/pull/406/files#r1176381729.

Reply:
Sure!